libraries/standard/http/src/http_client.c

```diff
@@ -245,6 +245,986 @@ static int findHeaderValueParserCallback( http_parser * pHttpParser,
+        pResponse->bodyLen = 0;
+    }
+
+    /* The next location to write. */
+
+    /* MISRA Rule 11.8 flags casting away the const qualifier in the pointer
+     * type. This rule is suppressed because when the body is of transfer
+     * encoding chunked, the body must be copied over the chunk headers that
+     * preceed it. This is done to have a contigous response body. This does
+     * affect future parsing as the changed segment will always be before the
+     * next place to parse. */
+    /* coverity[misra_c_2012_rule_11_8_violation] */
+    pNextWriteLoc = ( uint8_t * ) ( pResponse->pBody + pResponse->bodyLen );
+
+    /* If the response is of type Transfer-Encoding: chunked, then actual body
+     * will follow the the chunked header. This body data is in a later location
+     * and must be moved up in the buffer. When pLoc is greater than the current
+     * end of the body, that signals the parser found a chunk header. */
+
+    /* MISRA rule 18.3 glags the following as pLoc not pointing to the same
+     * object as pNextWriteLoc. This rule is suppressed because http-parser will
+     * always return a pLoc within the response buffer limit. */
+    /* coverity[misra_c_2012_rule_18_3_violation] */
+    if( pLoc > ( const char * ) pNextWriteLoc )
+    {
+        ( void ) memcpy( pNextWriteLoc, ( const uint8_t * ) pLoc, length );
+    }
+
+    /* Increase the length of the body found. */
+    pResponse->bodyLen += length;
+
+    /* Set the next location of parsing. */
+    pParsingContext->pBufferCur = pLoc + length;
+
+    LogDebug( ( "Response parsing: Found the response body. "
+                "BodyLength=%d",
+                length,
+                pLoc ) );
+
+    return shouldContinueParse;
+}
+
+/*-----------------------------------------------------------*/
+
+/* The MISRA directive 4.6 violation is for using primitive type int, instead of
+ * a typedef which denotes the signedness and size of the type. This violation
+ * is suppressed because this callback follows the function signature for the
+ * http_cb type in http-parser. */
+
+/* MISRA directive 8.13 flags that pHttpParser should be a const parameter since
+ * it does not change. This rule is suppressed because http-parser defines
+ * their callback function signatures without the const. */
+/* coverity[misra_c_2012_directive_4_6_violation] */
+/* coverity[misra_c_2012_rule_8_13_violation] */
+static int httpParserOnMessageCompleteCallback( http_parser * pHttpParser )
+{
+    HTTPParsingContext_t * pParsingContext = NULL;
+
+    assert( pHttpParser != NULL );
+
+    /* The MISRA rule 11.5 violation flags casting a void pointer to another
+     * type. This rule is suppressed here because the http-parser library
+     * requires that private context into callbacks must be set to a void
+     * pointer variable. */
+    /* coverity[misra_c_2012_rule_11_5_violation] */
+    pParsingContext = ( HTTPParsingContext_t * ) ( pHttpParser->data );
```

Comment:
assert for pHttpParser->data != NULL needed?

Reply:
yes missed thanks